src/main/kotlin/tdd/domain/Board.kt

```diff
@@ -0,0 +1,47 @@
+package tdd.domain
+
+class Board(
+    val cells: Map<Coordinate, Cell>
+) {
+    fun open(coordinate: Coordinate) {
+        open(ArrayDeque(listOf(coordinate)))
+    }
+
+    private tailrec fun open(coordinates: ArrayDeque<Coordinate>, visited: MutableSet<Coordinate> = mutableSetOf()) {
+        val coordinate = coordinates.removeFirstOrNull() ?: return
+        visited.add(coordinate)
+        val cell = cellOf(coordinate)
+        val aroundCoordinates = Coordinates.around(coordinate).filter { it in cells }
+        val aroundMineCount = aroundCoordinates.count { cellOf(it).isMine() }
+        cell.open(aroundMineCount)
+        if (cell.isZero()) {
+            coordinates.addAll(aroundCoordinates.filterNot { it in visited })
+        }
+
+        open(coordinates, visited)
+    }
+
+    fun hasMine(coordinate: Coordinate): Boolean = cellOf(coordinate).isMine()
+
+    fun isRunning(): Boolean {
+        val closedCount = cells.values.count { !it.isOpened() && !it.isMine() }
+        val mineCount = cells.values.count { it.isMine() }
+        return closedCount > mineCount
+    }
+
+    private fun cellOf(key: Coordinate): Cell {
+        return cells[key] ?: throw IllegalArgumentException("보드에 좌표가 존재하지 않습니다")
+    }
+
+    companion object {
+        fun create(height: Int, width: Int, mineCount: Int, randomCoordinates: (Int) -> Set<Coordinate>): Board {
+            val mineCoordinates = randomCoordinates(mineCount)
+            val coordinateToCells = Coordinates.all(height, width).map {
+                val cell = if (it in mineCoordinates) Cell(Mine) else Cell()
```

Comment:
```suggestion
                val cell = if (it in mineCoordinates) {
                    Cell(Mine) 
                } else {
                    Cell()
                }
```
if-else 문은 중괄호를 붙여주는 것이 컨벤션입니다. (이하 관련 내용 동일)

ref. https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements

Reply:
cell을 만드는 로직은 Cell과 관련있으니, Cell 클래스에서 하는 것이 좋지 않을까요? 🤔

Reply:
예리하게 잘 보셨네요! 👍 
> 한 줄 안에 간결하게 나타낼 수 있다면 중괄호를 과감하게 제거해도 좋을 것 같다는 의견인데 어떠신가요?

좋습니다! 💯